uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php

```diff
@@ -197,7 +197,7 @@ private static function unsetDefaultTypes() {
    */
   private static function validateParams($params) {
     if(!empty($params['add_public_holiday_to_entitlement'])) {
-      self::validateAddPublicHolidayToEntitlement();
+      self::validateAddPublicHolidayToEntitlement($params);
```

Comment:
since you are only using the ID parameter, there is no need to pass all the parameters to  validateAddPublicHolidayToEntitlement method and you can only pass the ID , another way is to create a class property called $_ID which store the ID for the leave type you are currently editing